File: 1653. Minimum Deletions to Make String Balanced/solution.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        n = len(s)
        min_deletions = 0
        b_count = 0

        for ch in s:
            if ch == "b":
                b_count += 1
            else:
                min_deletions = min(min_deletions + 1, b_count)

        return min_deletions

File: 1653. Minimum Deletions to Make String Balanced/Solution.java
class Solution {
    public int minimumDeletions(String s) {
        int n = s.length();
        int minDeletions = 0;
        int bCount = 0;

        for (int i = 0; i < n; i++) {
            if (s.charAt(i) == 'b') {
                bCount++;
            } else {
                minDeletions = Math.min(minDeletions + 1, bCount);
            }
        }

        return minDeletions;
    }
}
